src/cosl/interfaces/cluster.py

```diff
@@ -112,6 +114,8 @@ class ClusterProviderAppData(cosl.interfaces.utils.DatabagModel):
     ### worker node configuration
     worker_config: str
     """The whole worker workload configuration, whatever it is. E.g. yaml-encoded things."""
+    worker_config_version: Optional[str] = None
```

Comment:
I don't like that we're receiving a `workload_version`, but sending back a `worker_config_version`, although they're supposed to be identical.
I don't see a better alternative though, perhaps `workload_version_supported: true`? In the end the worker isn't going to do anything with that value: all we really want to know is if the coordinator was able to satisfy the version we requested.